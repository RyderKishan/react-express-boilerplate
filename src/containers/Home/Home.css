.Home {
  flex: 1;
  display: flex;
  padding: 1em;
  min-height: calc(100% - 4em);
  flex-direction: column;
}

.make-center {
  width: 100%;
  place-content: center;
  display: grid;
}

.Home > .actions {
  margin-bottom: 1em;
}

.Home > .actions > button {
  margin-right: 1em;
  color: white;
  background-color: #0D475E;
}

.Home > .results {
  flex: 1;
  display: flex;
}

.actions > button[disabled] {
 color: gray;
}

.results .postContainer {
  width: 100%;
  display: flex;
  flex-wrap: wrap;
}

.postContainer .MuiPaper-root {
  max-width: 345px;
  margin: 0.5em;
}

.make-center .MuiCircularProgress-root {
  color: #0D475E;
}