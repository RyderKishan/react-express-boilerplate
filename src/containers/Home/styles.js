import { makeStyles } from '@material-ui/core/styles';

const useStyles = makeStyles({
  root: {
    maxWidth: 345,
    margin: '8px',
  },
  media: {
    height: 140,
  },
  postContainer: {
    display: 'flex',
    flexWrap: 'wrap',
  },
});

export default useStyles;
