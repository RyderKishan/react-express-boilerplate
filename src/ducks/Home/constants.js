const GET_POSTS = 'Home/GET_POSTS';
const SET_POSTS = 'Home/SET_POSTS';

export default {
  GET_POSTS,
  SET_POSTS,
};
