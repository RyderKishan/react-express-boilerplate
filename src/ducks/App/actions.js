import * as Constants from './constants';

const getUserDetails = () => ({
  type: Constants.GET_USER_DETAILS,
});

const setUserDetails = (userDetails) => ({
  type: Constants.SET_USER_DETAILS,
  userDetails,
});

export {
  getUserDetails,
  setUserDetails,
};
