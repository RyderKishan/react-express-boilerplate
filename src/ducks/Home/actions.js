import * as Constants from './constants';

const getPosts = () => ({
  type: Constants.GET_POSTS,
});

const setPosts = (posts) => ({
  type: Constants.SET_POSTS,
  posts,
});

export {
  getPosts,
  setPosts,
};
