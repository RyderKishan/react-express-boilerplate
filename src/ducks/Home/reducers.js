import * as Constants from './constants';

const HomeReducer = (state = {}, action) => {
  switch (action.type) {
    case Constants.GET_POSTS:
      return {
        ...state,
        isPostFetching: true,
      };
    case Constants.SET_POSTS:
      return {
        ...state,
        isPostFetching: false,
        posts: action.posts,
      };
    default:
      return state;
  }
};

export default HomeReducer;
