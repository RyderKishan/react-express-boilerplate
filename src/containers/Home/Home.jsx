import React, { shallowEqual } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import Card from '@material-ui/core/Card';
import CircularProgress from '@material-ui/core/CircularProgress';
import CardActionArea from '@material-ui/core/CardActionArea';
import CardActions from '@material-ui/core/CardActions';
import CardContent from '@material-ui/core/CardContent';
import Button from '@material-ui/core/Button';
import Typography from '@material-ui/core/Typography';

import * as Actions from '../../ducks/Home/actions';
import * as Selectors from '../../ducks/Home/selectors';

import './Home.css';

const Home = () => {
  const dispatch = useDispatch();
  const getPosts = () => dispatch(Actions.getPosts());
  const setPosts = (posts) => dispatch(Actions.setPosts(posts));
  const { posts, isPostFetching } = useSelector((state) => ({
    posts: Selectors.posts(state),
    isPostFetching: Selectors.isPostFetching(state),
  }),
  shallowEqual);
  return (
    <div className="Home">
      <div className="actions">
        <Button
          id="fetch-results"
          disabled={isPostFetching}
          variant="contained"
          color="primary"
          onClick={getPosts}
        >
          Fetch Results
        </Button>
        <Button
          id="clear-results"
          disabled={posts && posts.length <= 0}
          variant="contained"
          color="primary"
          onClick={() => setPosts([])}
        >
          Clear Results
        </Button>
      </div>
      <div className="results">
        {
          !isPostFetching ? (
            <div className="postContainer">
              {
                posts && posts.length > 0 ? posts.map((post) => (
                  <Card key={post.id}>
                    <CardActionArea>
                      <CardContent>
                        <Typography gutterBottom variant="h5" component="h2">
                          {post.title}
                        </Typography>
                        <Typography variant="body2" color="textSecondary" component="p">
                          {post.body}
                        </Typography>
                      </CardContent>
                    </CardActionArea>
                    <CardActions>
                      <Button size="small" color="primary">
                        Share
                      </Button>
                      <Button size="small" color="primary">
                        Learn More
                      </Button>
                    </CardActions>
                  </Card>
                )) : (
                  <div className="make-center">
                    No Results
                  </div>
                )
              }
            </div>
          ) : (
            <div className="make-center">
              <CircularProgress id="posts-loading" />
            </div>
          )
        }
      </div>
    </div>
  );
};

export default Home;
