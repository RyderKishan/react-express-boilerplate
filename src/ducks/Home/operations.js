import * as Actions from './actions';

const dispatchGetPosts = (dispatch) => () => {
  dispatch(Actions.getPosts());
};

export default {
  dispatchGetPosts,
};
