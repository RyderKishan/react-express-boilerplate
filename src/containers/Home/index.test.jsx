import React from 'react';
import renderer from 'react-test-renderer';
// import { Provider } from 'react-redux';
import { shallow } from 'enzyme';

import Home from '.';
import * as Actions from '../../ducks/Home/actions';
// import mockStore from '../../stores/mockStore';
import Mocks from '../../../api/stubs/mocks';

// const store = mockStore({});

const mockDispatch = jest.fn();
mockDispatch.mockReturnValue(jest.fn());
const mockSelector = {
  posts: [],
  isPostFetching: false,
};

jest.mock('react-redux', () => ({
  useDispatch: () => mockDispatch,
  useSelector: () => mockSelector,
}));

describe('<Home /> - Functionality Test - Empty', () => {
  const wrapper = shallow(<Home />);
  it('Default Behaviour', () => {
    expect(wrapper.find('.Home').exists()).toBe(true);
    expect(wrapper.find('.Home .actions').exists()).toBe(true);
    expect(wrapper.find('.actions #fetch-results').exists()).toBe(true);
    expect(wrapper.find('.actions #clear-results').exists()).toBe(true);
    expect(
      wrapper.find('.Home .results .postContainer .make-center').text(),
    ).toBe('No Results');
  });
  it('Fetch Results', () => {
    wrapper.find('.actions #fetch-results').simulate('click');
    expect(mockDispatch).toHaveBeenCalledWith(Actions.getPosts());
  });
});

describe('<Home /> - Functionality Test - Loading', () => {
  mockSelector.isPostFetching = true;
  const wrapper = shallow(<Home />);
  it('Display Loader', () => {
    expect(
      wrapper.exists('.make-center #posts-loading'),
    ).toBe(true);
  });
});

describe('<Home /> - Functionality Test - Posts', () => {
  mockSelector.posts = Mocks.posts;
  mockSelector.isPostFetching = false;
  const wrapper = shallow(<Home />);
  it('Display Results', () => {
    expect(
      wrapper.find('.Home .results .postContainer').children().length,
    ).toBeGreaterThan(0);
  });
  it('Clear Results', () => {
    wrapper.find('.actions #clear-results').simulate('click');
    expect(mockDispatch).toHaveBeenCalledWith(Actions.setPosts([]));
  });
});

describe('<Home /> - Snapshot Tests', () => {
  mockSelector.posts = Mocks.posts;
  mockSelector.isPostFetching = false;
  const tree = renderer.create(<Home />).toJSON();
  it('Snapshot', () => {
    expect(tree).toMatchSnapshot();
  });
});
