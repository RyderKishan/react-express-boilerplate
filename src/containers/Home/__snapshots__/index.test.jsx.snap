// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> - Snapshot Tests Snapshot 1`] = `
<div
  className="Home"
>
  <div
    className="actions"
  >
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
      disabled={false}
      id="fetch-results"
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        Fetch Results
      </span>
    </button>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
      disabled={false}
      id="clear-results"
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        Clear Results
      </span>
    </button>
  </div>
  <div
    className="results"
  >
    <div
      className="postContainer"
    >
      <div
        className="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
      >
        <button
          className="MuiButtonBase-root MuiCardActionArea-root"
          disabled={false}
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <div
            className="MuiCardContent-root"
          >
            <h2
              className="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
            >
              optio molestias id quia eum
            </h2>
            <p
              className="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
            >
              quo et expedita modi cum officia vel magni
doloribus qui repudiandae
vero nisi sit
quos veniam quod sed accusamus veritatis error
            </p>
          </div>
          <span
            className="MuiCardActionArea-focusHighlight"
          />
        </button>
        <div
          className="MuiCardActions-root MuiCardActions-spacing"
        >
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-textSizeSmall MuiButton-sizeSmall"
            disabled={false}
            onBlur={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiButton-label"
            >
              Share
            </span>
          </button>
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-textSizeSmall MuiButton-sizeSmall"
            disabled={false}
            onBlur={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiButton-label"
            >
              Learn More
            </span>
          </button>
        </div>
      </div>
      <div
        className="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
      >
        <button
          className="MuiButtonBase-root MuiCardActionArea-root"
          disabled={false}
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <div
            className="MuiCardContent-root"
          >
            <h2
              className="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
            >
              in quibusdam tempore odit est dolorem
            </h2>
            <p
              className="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
            >
              itaque id aut magnam
praesentium quia et ea odit et ea voluptas et
sapiente quia nihil amet occaecati quia id voluptatem
incidunt ea est distinctio odio
            </p>
          </div>
          <span
            className="MuiCardActionArea-focusHighlight"
          />
        </button>
        <div
          className="MuiCardActions-root MuiCardActions-spacing"
        >
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-textSizeSmall MuiButton-sizeSmall"
            disabled={false}
            onBlur={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiButton-label"
            >
              Share
            </span>
          </button>
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-textSizeSmall MuiButton-sizeSmall"
            disabled={false}
            onBlur={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiButton-label"
            >
              Learn More
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
